item.rb

```diff
@@ -20,15 +20,27 @@ def add_genre(genre)
     genre.items.push(self) unless genre.items.include?(self)
   end
 
-  def add_label(label, set_label_items_variable: true)
+  def add_label(label, validation: true)
     @label = label
-    label.add_item(self) if set_label_items_variable
+    label.add_item(self) if validation
   end
 
   def move_to_archive
     @archive = true if can_be_archived?
   end
 
+  def to_json(*_args)
+    JSON.dump({
+                id: @id,
+                archived: @archived,
+                published_date: @published_date
+              })
+  end
+
+  def self.from_json(data)
+    new(data['published_date'], data['archived'])
+  end
+
```

Comment:
You could remove this method from the superclass because it's unused trough out the functionality of the app! You will only need it on the subclasses 👀